libs/wingc/src/type_check.rs

```diff
@@ -450,12 +450,60 @@ impl Subtype for Type {
 					parent_type.is_subtype_of(other)
 				})
 			}
-			(Self::Resource(res), Self::Interface(_)) => {
+			(Self::Resource(res), Self::Interface(iface)) => {
 				// If a resource implements the interface then it's a subtype of it (nominal typing)
-				res.implements.iter().any(|parent| {
+				let implements_iface = res.implements.iter().any(|parent| {
 					let parent_type: &Type = parent;
 					parent_type.is_subtype_of(other)
-				})
+				});
+
+				if implements_iface {
+					return true;
+				}
+
+				// To support flexible inflight closures, we say that any
+				// resource with an inflight method named "handle" is a subtype of
+				// any single-method interface with a matching "handle" method type.
+
+				// First, check if there is exactly one inflight method in the interface
+				if iface
+					.get_env()
+					.iter(true)
+					.filter(|(_name, kind, _info)| {
+						let type_ = kind.as_variable().expect("interface member should be a variable").type_;
+						type_.is_inflight_function()
+					})
+					.count() != 1
+				{
+					return false;
+				}
+
+				// Next, check that the interface's inflight method is named "handle" and get its type
+				let iface_handle_type = if let Some(method) = iface.get_env().try_lookup("handle", None) {
+					let method = method.as_variable().expect("interface member should be a variable");
+					if method.type_.is_inflight_function() {
+						method.type_
+					} else {
+						return false;
+					}
+				} else {
+					return false;
+				};
+
+				// Then get the type of the resource's "handle" method if it has one
+				let res_handle_type = if let Some(method) = res.get_env().try_lookup("handle", None) {
```

Comment:
again, might be easier using the `ClassLike` trait:
`res.get_method("handle".into())`